packages/block-editor/src/components/provider/index.native.js

```diff
@@ -19,7 +20,10 @@ import withRegistryProvider from './with-registry-provider';
 
 class BlockEditorProvider extends Component {
 	componentDidMount() {
-		this.props.updateSettings( this.props.settings );
+		this.props.updateSettings( {
+			...this.props.settings,
+			isRTL: I18nManager.isRTL,
+		} );
```

Comment:
@etoledom according to our conversation [on isRTL selector](https://github.com/WordPress/gutenberg/pull/21502#discussion_r407619227)  I have manage to add isRTL to settings in P`BLockEditorProvider` this way.

Using it in blocks is very readable and intuitive. Also we don't need to use different markup on web vs native site - it's cross platform after implement in this way. I have refactor other block to align with this - you can check this out.

However I think we need to confirm if it's good place to do this as you mentioned earlier.

Reply:
@tug could you please take a look at this bit?

Reply:
I believe it's the right approach if we need to update this setting dynamically from the editor itself.

What's missing right now though it the ability to "refresh" the editor when the settings change. If we look at the post editor we can see that the `EditorProvider` get the settings from the editor.
On the web, those settings are given from the PHP side of the plugin. So similarly we would need to pass it as a props from the parent apps.

Concerning `isRTL`, I guess we can use `I18nManager.isRTL` instead for now. What do you think about simply overwriting this setting in the native post editor for now:

At https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/editor.native.js#L47
have:
```
		settings = {
			...settings,
			isRTL: I18nManager.isRTL,
			hasFixedToolbar,
			focusMode,
		};
```

Reply:
Thanks for checking this. I think it make sense. Let me provide changes and well see how it works

Reply:
@Tug @etoledom I have applied changes mentioned above. It works fine so if you can taka a look one more time on this and do not have any other remarks I think we can merge